bundle/src/main/java/com/adobe/acs/commons/properties/impl/PropertyConfigServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2020 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.properties.impl;
+
+import com.adobe.acs.commons.properties.PropertyConfigService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+@Component(service = PropertyConfigService.class)
+@Designate(ocd = PropertyConfigServiceImpl.Config.class)
+public class PropertyConfigServiceImpl implements PropertyConfigService {
+
+    private List<Pattern> exclusionList;
+
+    @Override
+    public boolean isNotExcluded(final String propertyName) {
+        for (Pattern pattern : exclusionList) {
+            if (pattern.matcher(propertyName).matches()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isAllowedType(Object object) {
+        return String.class.equals(object.getClass()) || Long.class.equals(object.getClass());
```

Comment:
Any particular reason that values such as `Boolean` or `Date` are not allowable?

Reply:
I can see a use-case for Date/Calendar, and I think adding support for dates and date formatting would be a great future feature for this tool (unless we really want it included in this initial release), but I'm not sure I see the case for allowing people to add `true` or `false` to the content since it's going to be outputted as a string (i.e. can't use the boolean for logic).

Reply:
Sure, makes sense.  Date gets a bit tricky as that involves formatting as well.  Do we stick with this for now, having Date be a potential future addition?

Reply:
I think it's fine sticking with String/Long for now.  Date could be added later.